### ISSUE TITLE
overrideConfigFile isn't take the eslintrc config

### ISSUE THREAD

**Issue by RubensTen** [webpack-contrib/eslint-webpack-plugin#57]
Hello!
When I config the overrideConfigFile options it's seems like doesn't take  the config from eslintrc file,  
if I execute webpack, the plugin doesn't show the lint errors, but if I execute the eslint with the eslintrc config the scan show the errors correctly.

This is my webpack config:

```
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const ESLintPlugin = require('eslint-webpack-plugin');

module.exports = {
  plugins: [
    new HtmlWebpackPlugin({
      template: path.resolve(__dirname, 'src/index.html'),
      hash: true,
    }),
    new MiniCssExtractPlugin(),
    new ESLintPlugin({
      overrideConfigFile: path.resolve(__dirname, '.eslintrc.client.js'),
      context: path.resolve(__dirname, '/src'),
      files: path.resolve(__dirname, '**/*.jsx'),
      extensions: ['js', 'jsx'],
    }),
  ],
  resolve: {
    extensions: ['.js', '.jsx', '.json'],
  },
  devServer: {
    port: 3000,
    open: true,
    hot: true,
    historyApiFallback: true,
  },
};
```

These are my dependencies:
```
"devDependencies": {
    "@babel/core": "^7.12.10",
    "@babel/preset-env": "^7.12.10",
    "@babel/preset-react": "^7.12.10",
    "babel-loader": "^8.2.2",
    "css-loader": "^5.0.1",
    "eslint": "^7.2.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-airbnb-base": "^14.2.1",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-react": "^7.21.5",
    "eslint-plugin-react-hooks": "^4.0.0",
    "eslint-webpack-plugin": "^2.4.1",
    "file-loader": "^6.2.0",
    "html-webpack-plugin": "^4.5.0",
    "mini-css-extract-plugin": "^1.3.3",
    "webpack": "^5.10.1",
    "webpack-cli": "^4.2.0",
    "webpack-dev-server": "^3.11.0"
  },
```

eslint command:
`eslint ./src/**/*.jsx --config ./.eslintrc.client.js`

**Comment by codynova** [comment]
I think this is the same issue we're running into after trying to update `eslint-webpack-plugin` from version `2.1.0`. **All currently published versions after `2.1.0` do not work with the following config:**

```js
new ESLintPlugin({
  fix: true,
  files: [ './components', './pages' ],
  extensions: [ 'tsx', 'ts', 'jsx', 'js' ],
  overrideConfigFile: './.eslintrc',
  errorOnUnmatchedPattern: false,
})
```

```json
"devDependencies": {
  "@typescript-eslint/eslint-plugin": "4.13.0",
  "@typescript-eslint/parser": "4.13.0",
  "babel-eslint": "10.1.0",
  "babel-plugin-styled-components": "1.12.0",
  "eslint": "7.17.0",
  "eslint-plugin-react": "7.22.0",
  "eslint-plugin-react-hooks": "4.2.0",
  "eslint-webpack-plugin": "2.4.1"
}
```

Downgrading to `eslint-webpack-plugin@2.1.0` instantly fixes the issue, and upgrading to `eslint-webpack-plugin@2.2.0` causes the linter to stop working.

**Comment by ricardogobbosouza** [comment]
Hi @RubensTen 
Could you create a repository for reproduction?
Unfortunately you cannot reproduce the problem

**Comment by bradyclifford** [comment]
Having same issue.

**Comment by mingluzhang** [comment]
You can repro this issue from [this repo](https://github.com/mingluzhang/eslint-trouble-shoot.)
Just comment the ref of eslint-loader and non-comment the part for ESLintWebpackPlugin.

**Comment by RubensTen** [comment]
hi @codynova , @ricardogobbosouza this is the repo to reproduce the issue

https://github.com/RubensTen/webpack-eslint

Regards!

**Comment by codynova** [comment]
After more digging I found that my issue was caused by a change to the way filepaths work with the `files` option. Quoting my reply from above:

In my case, the issue was related to the `./` on the `files` option paths. The pattern `files: './pages'` worked fine in `v.2.1.0` but in `v2.2.0+` I simply needed to remove the `./` prefix:

```js
new ESLintPlugin({
  fix: true,
  files: [ 'components', 'pages' ],
  extensions: [ 'tsx', 'ts', 'jsx', 'js' ],
  overrideConfigFile: './.eslintrc',
  errorOnUnmatchedPattern: false,
})
```

**Comment by ricardogobbosouza** [comment]
Hi @RubensTen 

Your configuration is wrong:

```js
new ESLintPlugin ({
  eslintPath: require.resolve ('eslint'),
  overrideConfigFile: path.resolve (__ dirname, '.eslintrc.client.js'),
  context: path.resolve (__ dirname, 'src /'),
  // files: path.resolve (__ dirname, '** / *. tsx'), // files is relative a context
  extensions: ['tsx', 'ts'],
}),
```

See
https://github.com/webpack-contrib/eslint-webpack-plugin#files

**Comment by RubensTen** [comment]
@codynova @ricardogobbosouza my issue, the files path works for me :)

thank you!

**Comment by silkfire** [comment]
@ricardogobbosouza Is `overrideConfigFile` an officially supported option? I can't find it in the code or the documentation.

**Comment by ricardogobbosouza** [comment]
Hi @silkfire 
It is an option of the eslint itself
https://eslint.org/docs/developer-guide/nodejs-api#-new-eslintoptions
